src/Subscription.php

```diff
@@ -648,13 +710,17 @@ protected function mergeItemsThatShouldBeDeletedDuringSwap(Collection $items)
     {
         /** @var \Stripe\SubscriptionItem $stripeSubscriptionItem */
         foreach ($this->asStripeSubscription()->items->data as $stripeSubscriptionItem) {
-            $plan = $stripeSubscriptionItem->plan->id;
+            $plan = $stripeSubscriptionItem->plan;
 
-            if (! $item = $items->get($plan, [])) {
+            if (! $item = $items->get($plan->id, [])) {
                 $item['deleted'] = true;
+
+                if ($plan->usage_type == 'metered') {
+                    $item['clear_usage'] = true;
```

Comment:
What does this do?

Reply:
Clearing usage for a metered plan is required when the plan is deleted. Stripe has noted that this param will probably be removed in the future and the behavior would be the default.